api/envoy/api/v2/core/protocol.proto

```diff
@@ -24,6 +24,12 @@ message UpstreamHttpProtocolOptions {
   // upstream connections based on the downstream HTTP host/authority header, as seen by the
   // :ref:`router filter <config_http_filters_router>`.
   bool auto_sni = 1;
+
+  // Automatic validate upstream presented certificate for new upstream connections based on the
+  // downstream HTTP host/authority header, as seen by the
+  // :ref:`router filter <config_http_filters_router>`.
+  // This field is intended to set with `auto_sni` field.
```

Comment:
In general we almost always want this to be enabled with auto_sni is enabled, right? WDYT about making this a WKT bool which defaults to true when auto_sni is enabled, and then if false actually disables it? I could go either way on this but seems like we should in general try to be secure by default? WDYT?

Reply:
It depends, in case of dynamic forward proxy definitely yes and it will be default for forward proxy cluster. But in some other use case, like ours: is to having L1 envoy to L2 envoy have host-based SNI so SNI based routing (or filter chain match) can still work, but with internal certificates/identities.

Reply:
OK I see. OK this is fine if you think it's the best. We should probably have some arch/config docs on best practices? Maybe in the security and/or HTTP section?

/wait

Reply:
Added in SNI FAQ.

Reply:
The SAN that is validated here would be the same as the Authority field's value right? Out of ignorance, is it actually possible to pull out the SAN/SNI that is presented by the downstream to L1 and stick that into the upstream (to L2)?

Reply:
I also remembered one more caveat. If auto_host_rewrite is enabled at the cluster level, then the auto sni or auto san would certainly not match with the rewritten host value. (not sure if teh auto aspect takes the host_rewrite at route level into account). It would be useful to document this caveat in the FAQ or in the api field.

Reply:
> The SAN that is validated here would be the same as the Authority field's value right? Out of ignorance, is it actually possible to pull out the SAN/SNI that is presented by the downstream to L1 and stick that into the upstream (to L2)?

SAN is not presented to upstream, it is used to validate what certificate presented by upstream.